benchopt/utils/r_helpers.py

```diff
@@ -3,25 +3,28 @@
 # Make sure that R_HOME is loaded from the current interpreter to avoid
 # using the parent interpreter R_HOME in the sub-interpreter.
 if os.environ.get('R_HOME', None) is not None:
-    del os.environ['R_HOME']
+    os.environ.pop('R_HOME')
 
 
 import rpy2  # noqa: E402
 import rpy2.robjects.packages as rpackages  # noqa: E402
 try:
     from rpy2.robjects.packages import PackageNotInstalledError
+    import rpy2.situation
 except ImportError:
     # Backward compat for rpy2 version < 3.3
     try:
         from rpy2.rinterface_lib.embedded import \
             RRuntimeError as PackageNotInstalledError
+        import rpy2.situation
```

Comment:
If it is always present (not compat for <3.3), this should be imported only once before l.11

Reply:
It was introduced in 2.9 so I didn't know how far you wanted the compatibility.

Reply:
You call twice the same code so if this fails in l.13, this will also fail here (it is not a compat call here).
It should be imported in its own `try...except` to be compatible pior to 2.9 or simply called outside of the `try...except`.
I am in favor of the later but as you want.